src/test/java/edu/hm/hafner/analysis/ast/CheckStyleAstFactoryTest.java

```diff
@@ -0,0 +1,795 @@
+     */
+    private void checkThatHashesMatching(final String warningType, final String refactoring) {
+        checkThatHashesMatching(warningType, warningType, warningType, refactoring, true);
+    }
+
+    private void checkThatHashesNotMatching(final String warningType, final String refactoring) {
+        checkThatHashesMatching(warningType, warningType, warningType, refactoring, false);
+    }
+
+    /**
+     * Use this method for calculating the hashcode, if the filename before is different as after, disregarded that the
+     * after-filename has a postfix of the refactoring.
+     *
+     * @param warningType
+     *         the warningType
+     * @param beforeClass
+     *         the name of the class before
+     * @param afterClass
+     *         the name of the class after
+     * @param refactoring
+     *         the refactoring
+     * @param expectedEqualHashcode
+     *         <code>true</code>, if the expected hashcode is equal, otherwise <code>false</code>.
+     */
+    private void checkThatHashesMatching(final String warningType, final String beforeClass, final String afterClass,
+            final String refactoring, final boolean expectedEqualHashcode) {
+        String foldername = matchWarningTypeToFoldername(warningType);
+        long hashBefore = calcHashcode(beforeClass, foldername, true);
+        long hashAfter = calcHashcode(afterClass + refactoring, foldername, false);
+
+        if (expectedEqualHashcode) {
+            compareHashcode(hashBefore, hashAfter);
+        }
+        else {
+            compareHashcodeOnNonEquality(hashBefore, hashAfter);
+        }
+    }
+
+    private void checkThatHashesMatching(final String warningType, final String fileName, final String refactoring,
+            final boolean expectedEqualHashcode) {
+        checkThatHashesMatching(warningType, fileName, fileName, refactoring, expectedEqualHashcode);
+    }
+
+    private long calcHashcode(final String filename, final String foldername, final boolean beforeRefactoring) {
+        String javaFile = filename.concat(".java");
+        String xmlFile = filename.concat(".xml");
+
+        return calcHashcode(javaFile, foldername, xmlFile, beforeRefactoring);
+    }
+
+    private void compareHashcode(final long hashBefore, final long hashAfter) {
+        assertEquals("Hash codes don't match: ", hashBefore, hashAfter);
+    }
+
+    private void assertEquals(final String message, final long hashBefore, final long hashAfter) {
+        assertThat(hashBefore).as(message).isEqualTo(hashAfter);
+    }
+
+    private void assertEquals(final String message, final String hashBefore, final String hashAfter) {
+        assertThat(hashBefore).as(message).isEqualTo(hashAfter);
+    }
+
+    private void compareHashcodeOnNonEquality(final long hashBefore, final long hashAfter) {
+        getAssertNotEquals(hashBefore, hashAfter);
+    }
+
+    private void getAssertNotEquals(final long hashBefore, final long hashAfter) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Linguistics Antipattern - The getter 'getAssertNotEquals' should not return void linguistically](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=5019963)